queryfilter/datetimefilter.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import absolute_import
+
+import datetime
+
+from dateutil import parser
+import pytz
+
+from .base import FieldFilter, DictFilterMixin, DjangoQueryFilterMixin
+from .queryfilter import QueryFilter
+
+
+@QueryFilter.register_type_condition('datetime', 'datetime_range')
+class DatetimeRangeFilter(DjangoQueryFilterMixin, DictFilterMixin, FieldFilter):
+
+    @property
+    def start(self):
+        return get_start(self.filter_args.get("start"))
+
+    @property
+    def end(self):
+        return get_end(self.filter_args.get("end"))
+
+    def on_dicts(self, dicts):
+
+        def in_range(datum):
+            datetime_string = self.get(datum, self.field_name)
+            if isinstance(datetime_string, datetime.datetime):
+                to_compare = datetime_string
+            else:
+                to_compare = parse(datetime_string)
+            return self.start <= to_compare <= self.end
+
+        return list(filter(in_range, dicts))
+
+    def do_filter(self, queryset):
+        query_dict = {
+            "{}__gte".format(self.field_name): self.start,
```

Comment:
So we're going query on both gte & lte side, even if one of them is None.
How about only filter with valid input? it might cause some difference on performance?
Or ... it would be too minor to consider 🤔

Reply:
does replace None with max or min datetime object  means the same thing?

Reply:
yeah logically it does. But how about the implementation of database?

Reply:
I got your meaning!